tests/BenchmarkDotNet.IntegrationTests/InProcessTest.cs

```diff
@@ -0,0 +1,388 @@
+﻿using System;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Columns;
+using BenchmarkDotNet.Configs;
+using BenchmarkDotNet.Diagnosers;
+using BenchmarkDotNet.Jobs;
+using BenchmarkDotNet.Loggers;
+using BenchmarkDotNet.Running;
+using BenchmarkDotNet.Tests.Loggers;
+using BenchmarkDotNet.Toolchains.InProcess;
+using JetBrains.Annotations;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace BenchmarkDotNet.IntegrationTests
+{
+    public class InProcessTest : BenchmarkTestExecutor
+    {
+        public InProcessTest(ITestOutputHelper output) : base(output) { }
+
+        private const decimal DecimalResult = 42;
+        private const string StringResult = "42";
+
+        #region Direct runs
+        private const int UnrollFactor = 16;
+
+        [Fact]
+        public void TestBenchmarkActionSetup() => TestInvoke(x => BenchmarkAllCases.Setup(), UnrollFactor);
+        [Fact]
+        public void TestBenchmarkActionCleanup() => TestInvoke(x => x.Cleanup(), UnrollFactor);
+
+        [Fact]
+        public void TestBenchmarkActionVoid() => TestInvoke(x => x.InvokeOnceVoid(), UnrollFactor);
+        [Fact]
+        public void TestBenchmarkActionTask() => TestInvoke(x => x.InvokeOnceTaskAsync(), UnrollFactor, null);
+        [Fact]
+        public void TestBenchmarkActionRefType() => TestInvoke(x => x.InvokeOnceRefType(), UnrollFactor, StringResult);
+        [Fact]
+        public void TestBenchmarkActionValueType() => TestInvoke(x => x.InvokeOnceValueType(), UnrollFactor, DecimalResult);
```

Comment:
personally, I don't like to use [Facts] and not following the naming convention. For example `TestBenchmarkActionValueType` is not a fact, `BenchmarksReturningValueTypesAreSupported` is